### PR TITLE
Consumer API: Block new relationship when old one is in status deletion proposed

### PR DESCRIPTION
**Pull request by tnotheis**
# Readiness checklist

-   [x] I added/updated unit tests.
-   [ ] I added/updated integration tests.
-   [x] I ensured that the PR title is good enough for the changelog.
-   [x] I labeled the PR.

### CODE REVIEW

File: Modules/Relationships/src/Relationships.Domain/Aggregates/Relationships/Relationship.cs [hunk 1]
```diff
@@ -78,7 +78,7 @@ private static void EnsureTargetIsNotSelf(RelationshipTemplate relationshipTempl
 
     private static void EnsureNoOtherRelationshipToPeerExists(IdentityAddress target, IEnumerable<Relationship> existingRelationshipsToPeer)
     {
-        if (existingRelationshipsToPeer.Any(r => r.Status is RelationshipStatus.Active or RelationshipStatus.Pending or RelationshipStatus.Terminated))
+        if (existingRelationshipsToPeer.Any(r => r.Status is RelationshipStatus.Active or RelationshipStatus.Pending or RelationshipStatus.Terminated or RelationshipStatus.DeletionProposed))
```
Comment (jkoenig134) [inline Modules/Relationships/src/Relationships.Domain/Aggregates/Relationships/Relationship.cs:81]: can't this be simplified to Status != Rejected?

Comment (tnotheis) [reply Modules/Relationships/src/Relationships.Domain/Aggregates/Relationships/Relationship.cs:81]: No. There are also "Revoked" and "ReadyForDeletion".

**Review by jkoenig134 — APPROVED** [review]
LGTM :)

**Review by Dannyps — APPROVED** [review]

**Comment by tnotheis** [comment]
@britsta FYI

File: Modules/Relationships/src/Relationships.Domain/Aggregates/Relationships/Relationship.CreateTests.cs [hunk 2]
```diff
@@ -98,52 +98,56 @@ public void There_can_only_be_one_active_relationship_between_two_identities()
...
         var acting = () => new Relationship(RELATIONSHIP_TEMPLATE_OF_1, IDENTITY_2, DEVICE_2, null, existingRelationships);
 
         // Assert
-        acting.Should().Throw<DomainException>().WithError("error.platform.validation.relationshipRequest.relationshipToTargetAlreadyExists");
+        acting.Should().NotThrow<DomainException>();
     }
 
     [Fact]
-    public void A_new_relationship_can_be_created_after_decomposing_the_old_one()
+    public void Cannot_create_Relationship_if_terminated_Relationship_exists()
```
Comment (NikolaDmitrasinovic) [inline Modules/Relationships/src/Relationships.Domain/Aggregates/Relationships/Relationship.CreateTests.cs:139]: Wasn't this tested in the first test above?

Comment (tnotheis) [reply Modules/Relationships/src/Relationships.Domain/Aggregates/Relationships/Relationship.CreateTests.cs:139]: Oh man... I forgot to delete the old tests and rename the new one (it still contains "2" in its name)

**Pull request merged**